frontend/components/TeamsDropdown/TeamsDropdown.tsx

```diff
@@ -66,26 +86,179 @@ const TeamsDropdown = ({
     (option) => selectedTeamId === option.value
   )
     ? selectedTeamId
-    : teamOptions[0]?.value;
+    : teamOptions[0]?.value || 0; // Fallback to 0 or another default value
 
   const dropdownWrapperClasses = classnames(`${baseClass}-wrapper`, {
     disabled: isDisabled || undefined,
   });
 
+  const CustomDropdownIndicator = (
+    props: DropdownIndicatorProps<any, false, any>
+  ) => {
+    const { isFocused, selectProps } = props;
+    const color =
+      isFocused || selectProps.menuIsOpen
+        ? "core-fleet-blue"
+        : "core-fleet-black";
+
+    return (
+      <components.DropdownIndicator {...props} className={baseClass}>
+        <Icon
+          name="chevron-down"
+          color={color}
+          className={`${baseClass}__icon`}
+        />
+      </components.DropdownIndicator>
+    );
+  };
+
+  const customStyles: StylesConfig<INumberDropdownOption, false> = {
+    container: (provided) => ({
+      ...provided,
+      width: "80px",
+    }),
+    control: (provided, state) => ({
+      ...provided,
+      display: "flex",
+      flexDirection: "row",
+      width: "max-content",
+      padding: "8px 0",
+      backgroundColor: "initial",
+      border: 0,
+      boxShadow: "none",
+      cursor: "pointer",
+      "&:hover": {
+        boxShadow: "none",
+        ".team-dropdown__single-value": {
+          color: COLORS["core-vibrant-blue-over"],
+        },
+        ".team-dropdown__indicator path": {
+          stroke: COLORS["core-vibrant-blue-over"],
+        },
+      },
+      // When tabbing
+      "&.team-dropdown__control--is-focused": {
+        ".team-dropdown__single-value": {
+          color: COLORS["core-vibrant-blue-over"],
+        },
+        ".team-dropdown__indicator path": {
+          stroke: COLORS["core-vibrant-blue-over"],
+        },
+      },
+      // TODO: Figure out a way to apply separate &:focus-visible styling
```

Comment:
Does this TODO need to be resolved now? If not, can you mention it in an issue somewhere?

Reply:
Updated the comment